objects/src/block/header.rs

```diff
@@ -18,13 +19,13 @@ use crate::utils::serde::{
 /// - note_root is a commitment to all notes created in the current block.
 /// - batch_root is a commitment to a set of transaction batches executed as a part of this block.
 /// - proof_hash is a hash of a STARK proof attesting to the correct state transition.
-/// - version specifies the version of the protocol.
-/// - timestamp is the time when the block was created.
+/// - timestamp is the time when the block was created, in seconds since UNIX epoch.
 /// - sub_hash is a sequential hash of all fields except the note_root.
 /// - hash is a 2-to-1 hash of the sub_hash and the note_root.
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct BlockHeader {
+    version: u32,
     prev_hash: Digest,
```

Comment:
I wasn't actually thinking about moving the `version` field in the struct itself (only in serialization/deserialization). But not a strong preference from my end - so, if you think it is better to move it in the struct - that's fine too.

One other thing I just noticed - when serializing/deserializing, block header we seem to be serializing/deserializing `hash` and `sub_hash` field. These are computed fields, and I think it might be better not to serialize them to avoid potential inconsistencies. On deserialization we could use `BlockHeader::new()` to compute these field.